cognite_toolkit/_cdf_tk/prototypes/commands/dump_timeseries.py

```diff
@@ -0,0 +1,306 @@
+            for one_time_series in time_series_list:
+                write = one_time_series.as_write().dump(camel_case=True)
+                write.pop("parentId", None)
+                if "dataSetId" in write:
+                    data_set_id = write.pop("dataSetId")
+                    self._used_data_sets.add(data_set_id)
+                    write["dataSetExternalId"] = self._get_data_set_external_id(client, data_set_id)
+                if "assetId" in write:
+                    asset_id = write.pop("assetId")
+                    self._used_assets.add(asset_id)
+                    write["assetExternalId"] = self._get_asset_external_id(client, asset_id)
+                if expand_metadata and "metadata" in write:
+                    metadata = write.pop("metadata")
+                    for key, value in metadata.items():
+                        write[f"metadata.{key}"] = value
+                if "rootId" in write:
+                    root_id = write.pop("rootId")
+                    write["rootExternalId"] = self._get_time_series_external_id(client, root_id)
+                write_time_series.append(write)
+            yield write_time_series
+
+    def _get_time_series_external_id(self, client: CogniteClient, root_id: int) -> str:
+        if root_id in self.time_series_external_id_by_id:
+            return self.time_series_external_id_by_id[root_id]
+        try:
+            time_series = client.time_series.retrieve(id=root_id)
+        except CogniteAPIError as e:
+            raise ToolkitMissingResourceError(f"Failed to retrieve time_series {root_id}: {e}")
+        if time_series is None:
+            raise ToolkitMissingResourceError(f"TimeSeries {root_id} does not exist")
+        if not time_series.external_id:
+            raise ToolkitValueError(f"TimeSeries {root_id} does not have an external id")
+        self.time_series_external_id_by_id[root_id] = time_series.external_id
+        return time_series.external_id
+
+    def _get_data_set_external_id(self, client: CogniteClient, data_set_id: int) -> str:
+        if data_set_id in self.data_set_by_id:
+            return cast(str, self.data_set_by_id[data_set_id].external_id)
+        try:
+            data_set = client.data_sets.retrieve(id=data_set_id)
+        except CogniteAPIError as e:
+            raise ToolkitMissingResourceError(f"Failed to retrieve data set {data_set_id}: {e}")
+        if data_set is None:
+            raise ToolkitMissingResourceError(f"Data set {data_set_id} does not exist")
+        if not data_set.external_id:
+            raise ToolkitValueError(f"Data set {data_set_id} does not have an external id")
+        self.data_set_by_id[data_set_id] = data_set.as_write()
+        return data_set.external_id
+
+    def _get_asset_external_id(self, client: CogniteClient, asset_id: int) -> str:
+        if asset_id in self.asset_by_id:
+            return cast(str, self.asset_by_id[asset_id])
```

Comment:
Is this cast necessary? It seems the `asset_by_id` is typed correctly.
```suggestion
            return self.asset_by_id[asset_id]
```

Reply:
copy/pasted the `cast(str, self.data_set_by_id[data_set_id].external_id)` from dump_assets.py
- and I see now that `self.asset_by_id: dict[int, str] = {}` is explicit about the `str` type
- but `self.data_set_by_id: dict[int, DataSetWrite] = {}` not